sw/device/lib/dif/dif_spi_device.h

```diff
@@ -8,6 +8,10 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#ifdef MOCK_MMIO
```

Comment:
I think this `#ifdef` does not belong in `dif_spi_device.h`, and should be done in mmio